apps/hannk/interpreter/model.h

```diff
@@ -105,129 +105,121 @@ inline std::ostream &operator<<(std::ostream &s, const QuantizationInfo &q) {
     return s << "{" << q.scale << ", " << q.zero << ", " << q.dimension << "}";
 }
 
-inline Box without_strides(const std::vector<halide_dimension_t> &shape) {
-    Box result;
-    result.reserve(shape.size());
-    for (const halide_dimension_t &i : shape) {
-        result.emplace_back(i);
-    }
-    return result;
-}
-
 class Tensor {
     std::string name_;
     TensorType type_;
```

Comment:
Do we still need this? buffer().type() might suffice? It would lose support for string, complex, etc. but I'm not sure we need those anyways.

Reply:
Honestly, not sure. Adding support for those types would range form tricky-but-doable (for complex) to impossible (string). There's only a handful of places that inspect the type after creation. Let me try changing them.

Reply:
After looking a bit, I think the answer is that we can probably do away with hannk::TensorType entirely, and just use halide_type_t instead, but I think that would probably be cleaner to do in a subsequent PR.